### DISCUSSION TITLE
Bug: Eslint not working with  `eslint src/**/*.ts --fix`

### DISCUSSION THREAD

**Discussion by macartur** [eslint/eslint#18420] [Q&A]
### Environment

Node version: **v20.12.1**
npm version: **10.5.0**
Local ESLint version: **v8.57.0**
Operating System: Debian

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Way to reproduce</summary>

Create a file into `src/data/get_user/index.ts` with a linter error.
Try run the command bellow

```
npx eslint src/**/*ts
```

</details>

### What did you expect to happen?

A expect the file will be read and found typescript errors.

### What actually happened?

After run no error will be found because the file was not found.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IHF1b3RlczogW1wiZXJyb3JcIiwgXCJkb3VibGVcIl0gKi9cbmNvbnN0IGEgPSBcImJcIjsiLCJvcHRpb25zIjp7InJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjpbImVycm9yIl0sImZvci1kaXJlY3Rpb24iOlsiZXJyb3IiXSwiZ2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjpbImVycm9yIl0sIm5vLWNhc2UtZGVjbGFyYXRpb25zIjpbImVycm9yIl0sIm5vLWNsYXNzLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb21wYXJlLW5lZy16ZXJvIjpbImVycm9yIl0sIm5vLWNvbmQtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1iaW5hcnktZXhwcmVzc2lvbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1jb25kaXRpb24iOlsiZXJyb3IiXSwibm8tY29udHJvbC1yZWdleCI6WyJlcnJvciJdLCJuby1kZWJ1Z2dlciI6WyJlcnJvciJdLCJuby1kZWxldGUtdmFyIjpbImVycm9yIl0sIm5vLWR1cGUtYXJncyI6WyJlcnJvciJdLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tZHVwZS1lbHNlLWlmIjpbImVycm9yIl0sIm5vLWR1cGUta2V5cyI6WyJlcnJvciJdLCJuby1kdXBsaWNhdGUtY2FzZSI6WyJlcnJvciJdLCJuby1lbXB0eSI6WyJlcnJvciJdLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tZW1wdHktcGF0dGVybiI6WyJlcnJvciJdLCJuby1lbXB0eS1zdGF0aWMtYmxvY2siOlsiZXJyb3IiXSwibm8tZXgtYXNzaWduIjpbImVycm9yIl0sIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6WyJlcnJvciJdLCJuby1mYWxsdGhyb3VnaCI6WyJlcnJvciJdLCJuby1mdW5jLWFzc2lnbiI6WyJlcnJvciJdLCJuby1nbG9iYWwtYXNzaWduIjpbImVycm9yIl0sIm5vLWltcG9ydC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW52YWxpZC1yZWdleHAiOlsiZXJyb3IiXSwibm8taXJyZWd1bGFyLXdoaXRlc3BhY2UiOlsiZXJyb3IiXSwibm8tbG9zcy1vZi1wcmVjaXNpb24iOlsiZXJyb3IiXSwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tbmV3LW5hdGl2ZS1ub25jb25zdHJ1Y3RvciI6WyJlcnJvciJdLCJuby1ub25vY3RhbC1kZWNpbWFsLWVzY2FwZSI6WyJlcnJvciJdLCJuby1vYmotY2FsbHMiOlsiZXJyb3IiXSwibm8tb2N0YWwiOlsiZXJyb3IiXSwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjpbImVycm9yIl0sIm5vLXJlZGVjbGFyZSI6WyJlcnJvciJdLCJuby1yZWdleC1zcGFjZXMiOlsiZXJyb3IiXSwibm8tc2VsZi1hc3NpZ24iOlsiZXJyb3IiXSwibm8tc2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6WyJlcnJvciJdLCJuby1zcGFyc2UtYXJyYXlzIjpbImVycm9yIl0sIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjpbImVycm9yIl0sIm5vLXVuZGVmIjpbImVycm9yIl0sIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjpbImVycm9yIl0sIm5vLXVucmVhY2hhYmxlIjpbImVycm9yIl0sIm5vLXVuc2FmZS1maW5hbGx5IjpbImVycm9yIl0sIm5vLXVuc2FmZS1uZWdhdGlvbiI6WyJlcnJvciJdLCJuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmciOlsiZXJyb3IiXSwibm8tdW51c2VkLWxhYmVscyI6WyJlcnJvciJdLCJuby11bnVzZWQtcHJpdmF0ZS1jbGFzcy1tZW1iZXJzIjpbImVycm9yIl0sIm5vLXVudXNlZC12YXJzIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtYmFja3JlZmVyZW5jZSI6WyJlcnJvciJdLCJuby11c2VsZXNzLWNhdGNoIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtZXNjYXBlIjpbImVycm9yIl0sIm5vLXdpdGgiOlsiZXJyb3IiXSwicmVxdWlyZS15aWVsZCI6WyJlcnJvciJdLCJ1c2UtaXNuYW4iOlsiZXJyb3IiXSwidmFsaWQtdHlwZW9mIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
What does your eslint configuration look like? the "Link to Minimal Reproducible Example" seems not able to repro it.

**Comment by ValeraKaktotak** [comment]
I have the same problem, When i use  "eslint": "^7.32.0" in package.json everything is good. But when i use eslint": "^8.0.1 i have the same problem. Any Ignore rules i don't use. It happens when i use script - "lint": "eslint src/**/*.{tsx,ts}"

yarn run v1.22.19
$ eslint src/**/*.{tsx,ts}

Oops! Something went wrong! :(

ESLint: 8.57.0

You are linting "src/**/*.{tsx,ts}", but all of the files matching the glob pattern "src/**/*.{tsx,ts}" are ignored.

If you don't want to lint these files, remove the pattern "src/**/*.{tsx,ts}" from the list of arguments passed to ESLint.

If you do want to lint these files, try the following solutions:

* Check your .eslintignore file, or the eslintIgnore property in package.json, to ensure that the files are not configured to be ignored.
* Explicitly list the files from this glob that you'd like to lint on the command-line, rather than providing a glob as an argument.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by paulocoghi** [comment]
I'm having the same bug, or at least a similar one.

After updating from ESLint 8.57.0 to 9.0.0 and moving `.eslintrc.js` to `eslint.config.js` (without changing anything inside it), I received the error:

```
Oops! Something went wrong! :(

ESLint: 9.0.0

You are linting "{src,apps,libs,test}/**/*.ts", but all of the files matching the glob pattern "{src,apps,libs,test}/**/*.ts" are ignored.

If you don't want to lint these files, remove the pattern "{src,apps,libs,test}/**/*.ts" from the list of arguments passed to ESLint.

If you do want to lint these files, try the following solutions:

* Check your .eslintignore file, or the eslintIgnore property in package.json, to ensure that the files are not configured to be ignored.
* Explicitly list the files from this glob that you'd like to lint on the command-line, rather than providing a glob as an argument.
```

---

Here is my `eslint.config.js`:

```javascript
const isFixMode = process.argv.includes("--fix");

module.exports = {
  parser: "@typescript-eslint/parser",
  extends: [
    "xo",
    "xo-space",
    "xo-typescript",
    "prettier",
    "plugin:import/typescript",
    "plugin:unicorn/recommended",
    "plugin:prettier/recommended",
  ],
  ignorePatterns: ["**/dist/*", "**/node_modules/*"],
  overrides: [
    {
      plugins: ["jest"],
      files: [
        "test/**/*.ts",
        "tests/**/*.ts",
        "**/*.spec.ts",
      ],
      rules: {
        "@typescript-eslint/no-unsafe-assignment": "off",
        "max-nested-callbacks": "off",
        "jest/no-focused-tests": "error",
        ...(isFixMode && {
          "jest/no-focused-tests": "warn",
        }),
      },
    },
  ],
  parserOptions: {
    ecmaVersion: 2021,
    sourceType: "module",
  },
  env: {
    commonjs: true,
    es6: true,
    node: true,
    jest: true,
  },
  plugins: [
    "@typescript-eslint",
    "import",
    "unused-imports",
    "unicorn",
    "prettier",
  ],
  rules: {
    "prettier/prettier": ["error"],
    "@typescript-eslint/member-ordering": "off",
    "@typescript-eslint/no-unsafe-assignment": "off",
    "@typescript-eslint/no-unsafe-argument": "off",
    "@typescript-eslint/no-unsafe-call": "off",
    "@typescript-eslint/no-unsafe-return": "off",
    "no-console": "off",
    "@typescript-eslint/ban-types": "off",
    "comma-dangle": ["error", "always-multiline"],
    "unicorn/no-new-array": "off",
    "unicorn/no-reduce": "off",
    "unicorn/no-array-reduce": "off",
    "unicorn/prevent-abbreviations": [
      "error",
      {
        replacements: {
          props: false,
          args: false,
          params: false,
          env: false,
        },
      },
    ],
    "import/prefer-default-export": "off",
    "import/first": "error",
    "import/newline-after-import": "error",
    "import/no-duplicates": "error",
    "import/no-unresolved": "error",
    "import/order": [
      "error",
      {
        "newlines-between": "always",
        alphabetize: { order: "asc" },
        groups: ["builtin", "external", "internal", "parent", "sibling"],
      },
    ],
    "unused-imports/no-unused-imports-ts": "error",
    "unused-imports/no-unused-vars-ts": [
      "warn",
      {
        vars: "all",
        varsIgnorePattern: "^_",
        args: "after-used",
        argsIgnorePattern: "^_",
      },
    ],
    "@typescript-eslint/naming-convention": [
      "error",
      {
        selector: ["variable"],
        modifiers: ["const"],
        format: ["strictCamelCase", "StrictPascalCase", "UPPER_CASE"],
        leadingUnderscore: "allow",
      },
    ],
    "new-cap": "off",
  },
  settings: {
    "import/ignore": ["node_modules"],
    "import/parsers": {
      "@typescript-eslint/parser": [".ts", ".tsx"],
    },
    "import/resolver": {
      typescript: {
        alwaysTryTypes: true,
      },
    },
  },
};
```

**Comment by quellen-sol** [comment]
Just hit this as well. Here's a minimal reproduction: https://github.com/quellen-sol/eslint-bug-repro

Using this in WSL2 Ubuntu with node version `20.11.0`

**Comment by snigdha920** [comment]
I have the same issue on eslint v9.2.0. I don't have any eslintignore files or `eslintIgnore` config in my package.json, but I still get:

```
> eslint --config .eslintrc.js --quiet src/**/*.ts


Oops! Something went wrong! :(

ESLint: 9.2.0

You are linting "src/**/*.ts", but all of the files matching the glob pattern "src/**/*.ts" are ignored.

If you don't want to lint these files, remove the pattern "src/**/*.ts" from the list of arguments passed to ESLint.

If you do want to lint these files, try the following solutions:

* Check your .eslintignore file, or the eslintIgnore property in package.json, to ensure that the files are not configured to be ignored.
* Explicitly list the files from this glob that you'd like to lint on the command-line, rather than providing a glob as an argument.

 ELIFECYCLE  Command failed with exit code 2.
```

**Comment by aladdin-add** [comment]
Thanks, I was able to repro! It's because the config does not support flat configs, and not using `overrides` for ts files, however, it has been deprecated:
![image](https://github.com/eslint/eslint/assets/13050025/da88fb3c-01b2-4505-9b63-c5e87eb366fd)


a workaround:

```patch
diff --git a/eslint.config.mjs b/eslint.config.mjs
index 582d1ff..a033828 100644
--- a/eslint.config.mjs
+++ b/eslint.config.mjs
@@ -18,5 +18,5 @@ export default [
       "no-unused-vars": "error",
     },
   },
-  ...compat.extends("standard-with-typescript"),
+  ...compat.extends("standard-with-typescript").map(config => ({files: ["**/*.ts"], ...config})),
 ];
```

**Reply by codeef** [reply]
Hi @aladdin-add 
I have the same issue, tried to figure out what you meant with the workaround you showed here, but I'm not sure what I need to do because you're refering to the standard-with-typescript and I'm not using that

I have [this repo](https://github.com/codeef/eslint-try-0) with the exact same issue
It worked with the .eslintrc file perfectly and now with the eslint.config.js it doesn't
Can you please tell me what should I do to fix it?

**Reply by aladdin-add** [reply]
From `.eslintrc.*` to `eslint.config.js`, it's not just a matter of renaming files. please read the docs to do the migration: https://eslint.org/docs/latest/use/configure/migration-guide

**Comment by ExpliuM** [comment]
Same issue for me...

**Comment by nxxkap** [comment]
I was experiencing a similar error, but I was able to fix it by setting the environment variable `ESLINT_USE_FLAT_CONFIG` to `false`. However, the version of ESLint I am using is 8.5.7, and I recall that Flat Config became the default starting from version 9.0.0, so this is somewhat strange.

**Reply by nxxkap** [reply]
Also, try changing the file pattern you pass to the CLI from a glob pattern to `.`. This changed my error message from a strange one about ignore patterns to one about Flat Config.

**Comment by Yusuf007R** [comment]
same problem here

**Comment by 0ss** [comment]
Experiencing same problem here, all of the sudden happening 
 `"eslint": "^8.0.0"`

**Reply by sahin52** [reply]
Having "eslint": "^8.0.0" in package.json means install latest 8th version, and the latest 8.x.x version has that problem. Please either use "eslint": "8.0.0" without that mark, or update your packages from time to time

**Comment by SySagar** [comment]
for me this worked:

/** @type {import("eslint").Linter.Config} */
module.exports = {
  root: true,
  parser: '@typescript-eslint/parser',
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:react/recommended',
  ],
  plugins: ['@typescript-eslint', 'react'],
  parserOptions: {
    project: true,
    ecmaVersion: 'latest',
    sourceType: 'module',
  },
};


also I added it in my tsconfig like this:
`  "include": ["lib", "./lib/declaration.d.ts", ".eslintrc.cjs"],`

**Comment by EvgenyTomson** [comment]
Have you tried specifying the ESLint startup script in package.json as:
`"lint": "eslint \"src/**/*.ts\""`

**Comment by HopeKenga** [comment]
Has anyone been able to fix this? I'm experiencing the same even with all the provided solution

**Reply by yinsang** [reply]
https://github.com/eslint/eslint/issues/19361#issuecomment-2613131504
using --ext can run lint successfully, but if you want to lint ts file, you may need to use passing typescript config like https://www.npmjs.com/package/@typescript-eslint/eslint-plugin  to ESLint.
```
eslint --ext src/**/*.{tsx,ts}
```

**Reply by AR-CADE** [reply]
> Has anyone been able to fix this? I'm experiencing the same even with all the provided solution

back to tslint lol

**Comment by PFight** [comment]
Found solution for error `"You are linting "src/**/*.ts*", but all of the files matching the glob pattern "src/**/*.ts*" are ignored."` I have got this error when upgraded from old eslint and coverted old config to flat format. Solution is to add next row to config:

     "files": [ '**/*.ts', '**/*.tsx' ],

Old config worked without this row, but looks like new eslint requires this.
